### ISSUE TITLE
Azure | Marketplace ARM template fails deployment

### ISSUE THREAD

**Issue by AsafJosefMor** [nopSolutions/nopCommerce#5596]
nopCommerce Version: Since release-4.40.2

Steps to reproduce the problem: Deploy nopCommerce from Azure marketplace.

The returned state is `Conflict`.

**Comment by blueflightx7** [comment]
Reproducible for a week now. Anyone know if the ARM template is available publicly that is being used to deploy the solution via the Marketplace?

**Comment by AsafJosefMor** [comment]
Hi, any info on this?

Yes, the ARM template is available publicly.

https://portal.azure.com/#create/nopCommerce.nopCommerce

**Comment by hubby2004** [comment]
I can confirm that the azure deployment is not working. It terminates with status `Conflict`.

The first issue is that the website returns 500.32 which is fixed by changing the bitness from 32bit to 64bit. However, after that IIS returns 500.31 error:

```
HTTP Error 500.31 - ANCM Failed to Find Native Dependencies
Common solutions to this issue:
The specified version of Microsoft.NetCore.App or Microsoft.AspNetCore.App was not found.
Troubleshooting steps:
Check the system event log for error messages
Enable logging the application process' stdout messages
Attach a debugger to the application process and inspect
For more information visit: https://go.microsoft.com/fwlink/?LinkID=2028526 #
```

## System Information:

```powershell
PS C:\home> dotnet --info
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.201
 Commit:    a09bd5c86c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x86
 Base Path:   C:\Program Files (x86)\dotnet\sdk\5.0.201\

Host (useful for support):
  Version: 5.0.4
  Commit:  f27d337295

.NET SDKs installed:
  1.1.14 [C:\Program Files (x86)\dotnet\sdk]
  2.1.518 [C:\Program Files (x86)\dotnet\sdk]
  2.1.521 [C:\Program Files (x86)\dotnet\sdk]
  2.1.522 [C:\Program Files (x86)\dotnet\sdk]
  2.2.109 [C:\Program Files (x86)\dotnet\sdk]
  3.1.111 [C:\Program Files (x86)\dotnet\sdk]
  3.1.112 [C:\Program Files (x86)\dotnet\sdk]
  3.1.113 [C:\Program Files (x86)\dotnet\sdk]
  5.0.201 [C:\Program Files (x86)\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.26 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.14 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.26 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.14 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.3 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.13 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.16 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.13 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.26 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.14 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.3 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.13 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
```
@skoshelev can you please assist with this issue?

**Comment by skoshelev** [comment]
Hi, @hubby2004 and @AsafJosefMor.

Unfortunately, the package in the marketplace is assembled only for the win-x64 architecture and will not work in a 32-bit operating system or in an x86 architecture. The same goes for win-arm. To deploy a site on nopCommerce in a non-win-64 environment, you will have to rebuild it yourself. [This article](https://docs.nopcommerce.com/en/developer/tutorials/azure-publish.html) from our documentation might help you

closed #5596

**Comment by blueflightx7** [comment]
HI team, this is for deployment directly from the marketplace and not for deployment from a custom arm template. Deployments from the marktplace do not provide the option for some to choose the type of architecture to deploy on. Only App service, databse, and monitoring. This problem still exists

**Comment by AsafJosefMor** [comment]
Hi @blueflightx7 @skoshelev,

but what about the "HTTP Error 500.31 - ANCM Failed to Find Native Dependencies" which is not related and reproduces even with x64?

**Comment by AndreiMaz** [comment]
We'll check it

**Comment by hubby2004** [comment]
Thanks @AndreiMaz

**Comment by skoshelev** [comment]
> but what about the "HTTP Error 500.31 - ANCM Failed to Find Native Dependencies" which is not related and reproduces even with x64?

Hi. We fixed the problem with Azure Resource Manager templates (ARM templates) and now deployment works as expected. But we were still trying to found the reason of the 500.31 error

**Comment by AsafJosefMor** [comment]
![image](https://user-images.githubusercontent.com/20573715/129715060-96b5d19e-61ff-4e19-8da9-624c53ae6499.png)

**Comment by skoshelev** [comment]
We found the causes of error 500.31 and fixed it. The deployment and installation have now been fully working.

Closed #5596

**Comment by jjcowo** [comment]
Is the template still on the marketplace? I was unable to find it.
